### PR TITLE
Add elevation to Chips to allow for more flexibility

### PR DESCRIPTION
**Pull request by rami-a**
Expose elevation on chips and allow for it to be customized by the developer.

### CODE REVIEW

File: packages/flutter/lib/src/material/chip.dart [hunk 1]
```diff
@@ -111,6 +111,13 @@ abstract class ChipAttributes {
   ///
   ///  * [MaterialTapTargetSize], for a description of how this affects tap targets.
   MaterialTapTargetSize get materialTapTargetSize;
+
+  /// Elevation to be applied on the chip relative to its parent
```
Comment (johnsonmh) [inline packages/flutter/lib/src/material/chip.dart]: add a period .

File: packages/flutter/test/material/chip_test.dart [hunk 2]
```diff
@@ -1484,7 +1484,11 @@ void main() {
 
     final ChipThemeData chipTheme = theme.chipTheme;
 
-    InputChip inputChip = const InputChip(label: Text('Label'), pressElevation: 8.0);
+    InputChip inputChip = const InputChip(
+      label: Text('Label'),
+      pressElevation: 8.0,
```
Comment (johnsonmh) [inline packages/flutter/test/material/chip_test.dart]: We should test that when `elevation` and `pressElevation` are not supplied, the defaults are `0.0` and `8.0` respectively

File: packages/flutter/lib/src/material/chip.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/chip.dart]: Done

File: packages/flutter/test/material/chip_test.dart (hunk 2 above)
Comment (rami-a) [reply packages/flutter/test/material/chip_test.dart]: Good idea, done

**Review by johnsonmh — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
cc @goderbauer, are there any restrictions on elevation and pressElevation wrt scenic?

**Comment by goderbauer** [comment]
@jonahwilliams There shouldn't be an issue.

**Review by HansMuller — APPROVED** [review]
LGTM

Why isn't elevation part of ChipThemeData?

For chips with a pressElevation parameter it would be good to add: that "typically this value is greater than [elevation]".

**Comment by rami-a** [comment]
Thanks @HansMuller, I have a follow up PR that adds elevation and pressElevation to ChipThemeData coming soon. I broke it apart for simplicity.

I agree with that comment about pressElevation being higher and will make that tweak to the wording.

**Pull request merged**